### ISSUE TITLE
[Bug]: Keybind 'Button A' have no effects on x86

### ISSUE THREAD

**Issue by PrimaIVoid** [visualboyadvance-m/visualboyadvance-m#1061]
### Expected behavior?

Going to Menu:
`Options -> Input -> Configure -> Button A`
Trying to set the relative Button A with any key (Example: 'A', 'F', 'Y') should produce the correct ingame effect when pressed

### Actual behavior?

After setting the Button A with any key, pressing it ingame doesn't produce any effect
Toggling on the setting: `Options ->  Input -> Allow keyboard background input` make the corresponding keybind work as intended

After several testing this only seem to affect x86 (32bits) releases of VBA-M while x64 releases works as intended

### Steps to reproduce the problem

1. `Options -> Input -> Configure -> Button A`
2. Set Button A -> 'A'
3. Pressing 'A' ingame doesn't produce any effect

### Error messages?

No

### What game were you playing

Not specific to any ROM. Same behaviour on every ROM

### What settings are you using?

Video driver (Opengl)
Sound driver (SDL)
Scale Filter Used?: No

### Specifications and Build details

VBA-M: https://github.com/visualboyadvance-m/visualboyadvance-m/commit/561d6e1ab03fce79c40a8ac5d30593966f55e2d1
(Only seems to affect 32-bit releases)
OS: Linux - Debian 11 with Xorg
Emulator running: Wine 7.2.2
CPU: Intel i5
GPU: Nvidia 760
RAM: 4GB

### Copyright Compliance

- [X] There are no infringing files attached to this bug report.

**Comment by Squall-Leonhart** [comment]
This appears to have a missing environment interaction that means i am unable to reproduce it.

**Comment by rkitover** [comment]
I can reproduce this in wine-staging on gentoo, but not in real Windows.

**Comment by rkitover** [comment]
With `WINEDEBUG='warn+all'` I see these messages when I press the `'A'` key:

```console
0024:warn:menu:grab_menu_ptr invalid menu handle=00000000
0024:warn:menu:grab_menu_ptr invalid menu handle=00000000
```

**Comment by PrimaIVoid** [comment]
Casually playing a game in another emulator (PCSX2) i noticed i had a very similar problem... I opened an issue on their project and found out what the real problem is. I was said that some keyboards (real hardware) in my case in my laptop, may have a "**rollover limit** so pressing some combination of 3+ keys will not be registered, therefore ignored

In this website i was able to confirm my problem, being unable to do what other user with different keyboards was able to do:
https://www.mechanical-keyboard.org/key-rollover-test/

**Comment by rkitover** [comment]
Looks like this bug has been fixed in wine, just tried it in wine 9.0 in Ubuntu and it works.
